apis/compute/v1beta1/computefirewallpolicyrule_types.go

```diff
@@ -0,0 +1,160 @@
+	// +optional
+	Disabled *bool `json:"disabled,omitempty"`
+
+	/* Denotes whether to enable logging for a particular rule. If logging is enabled, logs will be exported to the configured export destination in Stackdriver. Logs may be exported to BigQuery or Pub/Sub. Note: you cannot enable logging on "goto_next" rules. */
+	// +optional
+	EnableLogging *bool `json:"enableLogging,omitempty"`
+
+	/* Immutable. */
+	FirewallPolicyRef *refs.ComputeFirewallPolicyRef `json:"firewallPolicyRef"`
+
+	/* A match condition that incoming traffic is evaluated against. If it evaluates to true, the corresponding 'action' is enforced. */
+	Match *FirewallpolicyruleMatch `json:"match"`
+
+	/* Immutable. An integer indicating the priority of a rule in the list. The priority must be a positive value between 0 and 2147483647. Rules are evaluated from highest to lowest priority where 0 is the highest priority and 2147483647 is the lowest prority. */
+	Priority int64 `json:"priority"`
+
+	// +optional
+	TargetResources []*refs.ComputeNetworkRef `json:"targetResources,omitempty"`
+
+	// +optional
+	TargetServiceAccounts []*refs.IAMServiceAccountRef `json:"targetServiceAccounts,omitempty"`
+}
+
+// +kcc:proto=google.cloud.compute.v1.FirewallPolicyRule
+type ComputeFirewallPolicyRuleStatus struct {
+	commonv1alpha1.CommonStatus `json:",inline"`
+	/* Type of the resource. Always `compute#firewallPolicyRule` for firewall policy rules */
+	// +optional
+	Kind *string `json:"kind,omitempty"`
+
+	/* Calculation of the complexity of a single firewall policy rule. */
+	// +optional
+	RuleTupleCount *int64 `json:"ruleTupleCount,omitempty"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:categories=gcp,shortName=gcpcomputefirewallpolicyrule;gcpcomputefirewallpolicyrules
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="cnrm.cloud.google.com/dcl2crd=true";"cnrm.cloud.google.com/managed-by-kcc=true";"cnrm.cloud.google.com/stability-level=stable";"cnrm.cloud.google.com/system=true"
+// +kubebuilder:printcolumn:name="Age",JSONPath=".metadata.creationTimestamp",type="date"
+// +kubebuilder:printcolumn:name="Ready",JSONPath=".status.conditions[?(@.type=='Ready')].status",type="string",description="When 'True', the most recent reconcile of the resource succeeded"
+// +kubebuilder:printcolumn:name="Status",JSONPath=".status.conditions[?(@.type=='Ready')].reason",type="string",description="The reason for the value in 'Ready'"
+// +kubebuilder:printcolumn:name="Status Age",JSONPath=".status.conditions[?(@.type=='Ready')].lastTransitionTime",type="date",description="The last transition time for the value in 'Status'"
+
+// ComputeFirewallPolicyRule is the Schema for the compute API
+// +k8s:openapi-gen=true
+type ComputeFirewallPolicyRule struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ComputeFirewallPolicyRuleSpec   `json:"spec,omitempty"`
```

Comment:
(this probably needs to come from the template code but we should `//+require` the spec here

Reply:
I actually copied the existing types because this is Beta resource. While I do not see `//+required` in the [template](https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/67b69fdebb742db5390e6b328d00c396765c9da9/dev/tools/controllerbuilder/template/apis/types.go#L90), do we have pending changes to the template recently?